log.h

```diff
@@ -20,55 +20,46 @@
 #include <errno.h>
 #include <stdarg.h>
 #include <stdlib.h>
-#include <syslog.h>
 
 #include "macro.h"
 
-#ifdef __cplusplus
-extern "C" {
-#endif
+class Log {
+public:
+    enum class Level {
+        ERROR = 0,
+        WARNING,
+        NOTICE,
+        INFO,
+        DEBUG,
+    };
 
-int log_open(void);
-int log_close(void);
+    static int open();
+    static int close();
 
-int log_get_max_level(void) _pure_;
-void log_set_max_level(int level);
-int log_internal(int level, int error,
-                 const char *file, int line,
-                 const char *format, ...) _printf_format_(5, 6);
+    static Level get_max_level() _pure_ { return _max_level; }
+    static void set_max_level(Level level);
 
-#define log_full_errno(level, error, ...)                               \
-    ({                                                                  \
-         int _level = (level), _e = (error);                            \
-         (log_get_max_level() >= LOG_PRI(_level))                       \
-            ? log_internal(_level, _e, __FILE__, __LINE__, __VA_ARGS__) \
-            : -abs(_e);                                                 \
-        })
+    static void logv(Level level, const char *format, va_list ap);
+    static void log(Level level, const char *format, ...) _printf_format_(2, 3);
```

Comment:
I think it's a matter of preference... I don't like the << operator since for my use cases the log is always a string not a function. Often logging is confused as normal output, or things that should go into dbg() (for which ifdef'ing by LOG_DEBUG would accomplish the same result).
If the implementation is reasonable, I wouldn't oppose to it, as long as it keep the properties of this lightweight logging. Feel free to provide an additional method for dealing with object logging.